### PR TITLE
Can no longer place an infinite quantity of items inside a microwave

### PR DESCRIPTION
**Pull request by SpeltIncorrectyl**
## About the PR
Fixes issue #23462

Microwaves now have an item capacity of 10 items. No more items can be inserted once this item limit has been reached.
10 items is more than enough for any recipe. The largest recipe is `RecipeStewSoup`, which needs 7 solids + 1 container for reagents (water).

## Why / Balance
Crates and lockers have finite inventory capacity. Microwaves, which shouldn't be used for item storage, and infact should be quite bad at it, can store an unlimited quantity of items. This does not make sense.

## Technical details
MicrowaveComponent has new integer field called Capacity.
MicrowaveSystem will compare the number of items already inside to the capacity, and will not insert any more unless the number of items inside is less than the capacity.
KitchenMicrowave prototype has this field set to 10.

## Media
https://github.com/space-wizards/space-station-14/assets/66873282/2c9ca5de-82c7-4972-a686-d6ec3c44ac26

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
MicrowaveComponent has new integer field Capacity.

**Changelog**
:cl:
- fix: Due to budget constraints, microwaves can no longer contain an infinite number of items. Now the limit is 10.

### CODE REVIEW

**Comment by GreyMario** [comment]
And what's the cook time on that stew?

**Comment by SpeltIncorrectyl** [comment]
> And what's the cook time on that stew?

10

**Comment by EmoGarbage404** [comment]
10 capacity is perfectly reasonable

File: Content.Server/Kitchen/Components/MicrowaveComponent.cs [hunk 1]
```diff
@@ -66,6 +66,9 @@ public sealed partial class MicrowaveComponent : Component
         public int CurrentCookTimeButtonIndex;
 
         public Container Storage = default!;
+
+        [DataField("capacity")]
```
Comment (EmoGarbage404) [inline Content.Server/Kitchen/Components/MicrowaveComponent.cs:70]: Remove the string and add a VV attribute

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 2]
```diff
@@ -260,9 +261,16 @@ private void OnInteractUsing(Entity<MicrowaveComponent> ent, ref InteractUsingEv
...
+            if (ent.Comp.Storage.Count < ent.Comp.Capacity)
+            {
+                args.Handled = true;
+                _handsSystem.TryDropIntoContainer(args.User, args.Used, ent.Comp.Storage);
+                UpdateUserInterfaceState(ent, ent.Comp);
+            }
+            else
+            {
+                _popupSystem.PopupEntity(Loc.GetString("microwave-component-interact-full"), ent, args.User);
+            }
```
Comment (EmoGarbage404) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:273]: Writing this as a guard clause would make this clearer.

**Review by EmoGarbage404 — APPROVED** [review]

**Comment by FairlySadPanda** [comment]
TARDIS microwave as an upgrade option?

**Comment by deltanedas** [comment]
bluespace microwave, syndicate potted plant but drag yourself in to teleport

**Comment by Froozigiusz** [comment]
> You've essentially removed the ability to cook three of those stews in one batch by adding three copies of the ingredients and setting the cook time to 30 seconds.

Although an interesting mechanic, I do see it as an exploit ngl.

**Comment by GreyMario** [comment]
> > You've essentially removed the ability to cook three of those stews in one batch by adding three copies of the ingredients and setting the cook time to 30 seconds.
> 
> Although an interesting mechanic, I do see it as an exploit ngl.

It's literally programmed to do exactly this on purpose? How does that make it an exploit.

**Comment by HoidC** [comment]
> > You've essentially removed the ability to cook three of those stews in one batch by adding three copies of the ingredients and setting the cook time to 30 seconds.
> 
> Although an interesting mechanic, I do see it as an exploit ngl.

how is it an exploit to cook more of a food that can be scaled up to cook more in one batch irl

**Comment by EmoGarbage404** [comment]
The exploit is sticking a million things on a microwave. Its reasonable to stick a few items but being unable to stick enough ingredients to make 3 instances of a recipe is a completely reasonable limitation

**Pull request merged**